x-pack/platform/plugins/shared/stack_connectors/public/common/auth/auth_config.test.tsx

```diff
@@ -480,4 +482,218 @@ describe('AuthConfig renders', () => {
+        expect(onSubmit).toHaveBeenCalled();
+      });
+
+      expect(onSubmit).toHaveBeenCalledWith({
+        data: {}, // Data is empty because form is invalid
+        isValid: false,
+      });
+    });
+
+    it('validates additionalFields input for valid/invalid JSON', async () => {
+      const testFormData = {
+        config: {
+          hasAuth: true,
+          authType: AuthType.OAuth2ClientCredentials,
+          accessTokenUrl: 'https://test.url',
+          clientId: 'testClient',
+        },
+        secrets: {
+          clientSecret: 'testSecret',
+        },
+      };
+
+      render(
+        <AuthFormTestProvider defaultValue={testFormData} onSubmit={onSubmit}>
+          <AuthConfig readOnly={false} isOAuth2Enabled={true} />
+        </AuthFormTestProvider>
+      );
+
+      let additionalFieldsInput: HTMLTextAreaElement | null = null;
+      await waitFor(() => {
+        additionalFieldsInput = document.querySelector('textarea');
+        expect(additionalFieldsInput).toBeInTheDocument();
+      });
+
+      expect(additionalFieldsInput).not.toBeNull();
+
+      await userEvent.clear(additionalFieldsInput!);
+      await userEvent.type(additionalFieldsInput!, '{{key": "value');
+
+      expect(await screen.findByText('Invalid JSON')).toBeInTheDocument();
+
+      await userEvent.clear(additionalFieldsInput!);
+      await userEvent.type(additionalFieldsInput!, '{{"sdf": "value"}');
+
+      await waitFor(() => {
+        expect(screen.queryByText('Invalid JSON')).not.toBeInTheDocument();
+      });
+    });
+
+    it('renders OAuth2 fields as readOnly when readOnly prop is true', async () => {
+      const initialJson = JSON.stringify({ initial: 'value' });
+      const testFormData = {
+        config: {
+          hasAuth: true,
+          authType: AuthType.OAuth2ClientCredentials,
+          accessTokenUrl: 'https://initial.url',
+          clientId: 'initialClient',
+          scope: 'readScope',
+          additionalFields: initialJson,
+        },
+        secrets: {
+          clientSecret: 'initialSecret',
+        },
+      };
+
+      render(
+        <AuthFormTestProvider defaultValue={testFormData} onSubmit={onSubmit}>
+          <AuthConfig readOnly={true} isOAuth2Enabled={true} />
+        </AuthFormTestProvider>
+      );
+
+      const accessTokenInput = await screen.findByTestId('accessTokenUrlAOuth2');
+      expect(accessTokenInput).toHaveAttribute('readonly');
+
+      const clientIdInput = await screen.findByTestId('clientIdOAuth2');
+      expect(clientIdInput).toHaveAttribute('readonly');
+
+      const clientSecretContainer = await screen.findByTestId('clientSecretOAuth2');
+      expect(clientSecretContainer).toHaveAttribute('readonly');
+
+      const scopeInput = await screen.findByTestId('ScopeOAuth2');
+      expect(scopeInput).toHaveAttribute('readonly');
+
+      const additionalFieldsContainer = await screen.findByTestId(
+        'additional_fieldsJsonEditorReadOnly'
```

Comment:
couldn't find a better way to test if the editor is in readonly mode. Couldn't find a class or something I could use to detect that it's readonly so I'm updating the test-subj-id to readonly the same time readonly is true and passed down to the component we do not own